sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -400,11 +491,58 @@ case class GpuCast(
     }
 
     withResource(minValue) { minValue =>
-      throwIfAny(values.lessThan(minValue))
+      throwIfAny(inclusiveMin match {
+        case true => values.lessThan(minValue)
+        case false => values.lessOrEqualTo(minValue)
+      })
     }
 
     withResource(maxValue) { maxValue =>
-      throwIfAny(values.greaterThan(maxValue))
+      throwIfAny(inclusiveMax match {
+        case true => values.greaterThan(maxValue)
+        case false => values.greaterOrEqualTo(maxValue)
+      })
+    }
+  }
+
+  /**
+   * Detects outlier values of a column given with specific range, and replaces them with
+   * a inputted substitution value.
+   *
+   * @param values ColumnVector
+   * @param minValue Named parameter for function to create Scalar representing range minimum value
+   * @param maxValue Named parameter for function to create Scalar representing range maximum value
+   * @param replaceValue Named parameter for function to create scalar to substitute outlier value
+   * @param inclusiveMin Whether the min value is included in the valid range or not
+   * @param inclusiveMax Whether the max value is included in the valid range or not
+   */
+  private def replaceOutOfRangeValues(values: ColumnVector,
+    minValue: => Scalar,
+    maxValue: => Scalar,
+    replaceValue: => Scalar,
+    inclusiveMin: Boolean = true,
+    inclusiveMax: Boolean = true): ColumnVector = {
+
+    withResource(minValue) { minValue =>
+      withResource(maxValue) { maxValue =>
+        val minPredicate = inclusiveMin match {
```

Comment:
nit: technically if we get an exception when calculating `maxPredicate`, `minPredicate` will leak.  Not a huge deal but it might be nice to fix it.

```
def calcMaxPred = inclusiveMax match {
  case true => values.greaterThan(maxValue)
  case false => values.greaterOrEqualTo(maxValue)
}

def calcMinPred = inclusiveMin match {
  case true => values.lessThan(minValue)
  case false => values.lessOrEqualsTo(minValue)
}

val rangePredicate = withResource(calcMinPred) { minPredicate =>
    withResource(calcMaxPred) { maxPredicate =>
      ...
    }
}
```

Reply:
Fixed